nodestats.go

```diff
@@ -0,0 +1,106 @@
+package mir
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/mir/pkg/logging"
+	t "github.com/filecoin-project/mir/pkg/types"
+	"github.com/filecoin-project/mir/pkg/util/maputil"
+)
+
+// ==============================================================================================================
+// Event dispatching statistics
+// ==============================================================================================================
+
+// eventDispatchStats saves statistical information about the dispatching of events between modules,
+// such the numbers of events dispatched for each module.
+type eventDispatchStats struct {
+	dispatchCounts map[t.ModuleID]int
+	eventCounts    map[t.ModuleID]int
+	numDispatches  int
+	lastDispatch   time.Time
+}
+
+// newDispatchStats returns a new eventDispatchStats object with all counters set to 0
+// and the last dispatch time set to the current time.
+func newDispatchStats(moduleIDs []t.ModuleID) eventDispatchStats {
+	stats := eventDispatchStats{
+		dispatchCounts: make(map[t.ModuleID]int),
+		eventCounts:    make(map[t.ModuleID]int),
+		numDispatches:  0,
+		lastDispatch:   time.Now(),
+	}
+	for _, moduleID := range moduleIDs {
```

Comment:
This is not necessary, accessing a key in a map that is not there will return the [zero value](https://go.dev/doc/effective_go#maps).

Reply:
Also the make already resets the map to a whole new instance (I saw it is used during `flushStats`.

Reply:
Actually, I didn't have it there before and added it later deliberately.
The point is that when I'm printing the stats (by looping through the map), I want all modules to be printed, even if they have been given 0 events.
In the last commit, a changed the printing even further, so the output is always consistently sorted by module ID.